### PR TITLE
Calibrate deployment [pipeline cluster]

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**

- [x] Configuration for deploying an instance of the calibrate microservice on the pipeline cluster.
- [x] Renames the workflow trigger file to reduce ambiguity.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-1040](https://airqoteam.atlassian.net/browse/PLAT-1040)

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
All calibrate endpoints through the pipeline cluster

**_ARE THERE ANY RELATED PRs?_**
 #755

### CODE REVIEW

**Review by 123MwanjeMike — APPROVED** [review]

File: .github/workflows/deploy-apis-to-staging.yml [hunk 1]
```diff
@@ -753,39 +753,58 @@ jobs:
...
-
-      ### run unit tests ###
-      
-      - name: Login to GCR
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (NoahNsimbe) [inline .github/workflows/deploy-apis-to-staging.yml:761]: Added docker Buildx to improve build performance

File: .github/workflows/deploy-apis-to-staging.yml [hunk 2]
```diff
@@ -753,39 +753,58 @@ jobs:
...
+          kubectl apply -f k8s/calibrate/stage-calibrate-api.yaml -n staging
+          kubectl set image deployment/airqo-stage-calibrate-api calibrate-pickle=${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-pickle-file:${{ needs.image-tag.outputs.build_id }} -n staging
+          kubectl set image deployment/airqo-stage-calibrate-api sta-calibrate=${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-api:${{ needs.image-tag.outputs.build_id }} -n staging
+          kubectl annotate deployment/airqo-stage-calibrate-api kubernetes.io/change-cause="Image updated to ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-api:${{ needs.image-tag.outputs.build_id }} on ${{ needs.image-tag.outputs.datetime }}" -n staging
+      -
+        name: Login to Pipeline k8s cluster
+        uses: azure/k8s-set-context@v1
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.K8S_PIPELINE_CONFIG }}
```
Comment (NoahNsimbe) [inline .github/workflows/deploy-apis-to-staging.yml:803]: Configuration for pipeline cluster

File: .github/workflows/deploy-apis-to-staging.yml [hunk 3]
```diff
@@ -753,39 +753,58 @@ jobs:
...
-          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-api:latest
-          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-pickle-file:latest
-      - name: Deploy to K8S
+          kubectl apply -f k8s/calibrate/stage-calibrate-api.yaml -n staging
+          kubectl set image deployment/airqo-stage-calibrate-api calibrate-pickle=${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-pickle-file:${{ needs.image-tag.outputs.build_id }} -n staging
+          kubectl set image deployment/airqo-stage-calibrate-api sta-calibrate=${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-api:${{ needs.image-tag.outputs.build_id }} -n staging
+          kubectl annotate deployment/airqo-stage-calibrate-api kubernetes.io/change-cause="Image updated to ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-stage-calibrate-api:${{ needs.image-tag.outputs.build_id }} on ${{ needs.image-tag.outputs.datetime }}" -n staging
+      -
+        name: Login to Pipeline k8s cluster
+        uses: azure/k8s-set-context@v1
```
Comment (Baalmart) [inline .github/workflows/deploy-apis-to-staging.yml:800]: curious about this `azure `mention here. @NoahNsimbe

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**